FirebaseRemoteConfig/SwiftNew/RemoteConfig.swift

```diff
@@ -0,0 +1,942 @@
+
+  // MARK: - Public Initializers and Accessors
+
+  @objc public static func remoteConfig(with app: FirebaseApp) -> RemoteConfig {
+    return remoteConfig(withFIRNamespace: RemoteConfigConstants.NamespaceGoogleMobilePlatform,
+                        app: app)
+  }
+
+  @objc public static func remoteConfig() -> RemoteConfig {
+    guard let app = FirebaseApp.app() else {
+      fatalError("The default FirebaseApp instance must be configured before the " +
+        "default Remote Config instance can be initialized. One way to ensure " +
+        "this is to call `FirebaseApp.configure()` in the App Delegate's " +
+        "`application(_:didFinishLaunchingWithOptions:)` or the `@main` struct's " +
+        "initializer in SwiftUI.")
+    }
+    return remoteConfig(withFIRNamespace: RemoteConfigConstants.NamespaceGoogleMobilePlatform,
+                        app: app)
+  }
+
+  @objc(remoteConfigWithFIRNamespace:)
+  public static func remoteConfig(withFIRNamespace firebaseNamespace: String) -> RemoteConfig {
+    guard let app = FirebaseApp.app() else {
+      fatalError("The default FirebaseApp instance must be configured before the " +
+        "default Remote Config instance can be initialized. One way to ensure " +
+        "this is to call `FirebaseApp.configure()` in the App Delegate's " +
+        "`application(_:didFinishLaunchingWithOptions:)` or the `@main` struct's " +
+        "initializer in SwiftUI.")
+    }
+
+    return remoteConfig(withFIRNamespace: firebaseNamespace, app: app)
+  }
+
+  // Use the provider to generate and return instances of FIRRemoteConfig for this specific app and
+  // namespace. This will ensure the app is configured before Remote Config can return an instance.
+  @objc(remoteConfigWithFIRNamespace:app:)
+  public static func remoteConfig(withFIRNamespace firebaseNamespace: String = RemoteConfigConstants
+    .NamespaceGoogleMobilePlatform,
+    app: FirebaseApp) -> RemoteConfig {
+    let provider = ComponentType<RemoteConfigInterop>
+      .instance(
+        for: RemoteConfigInterop.self,
+        in: app.container
+      ) as! any RemoteConfigProvider as RemoteConfigProvider
+    return provider.remoteConfig(forNamespace: firebaseNamespace)!
+  }
+
+  /// Last successful fetch completion time.
+  @objc public var lastFetchTime: Date? {
+    var fetchTime: Date?
+    queue.sync {
+      let lastFetchTimeInterval = self.settings.lastFetchTimeInterval
+      if lastFetchTimeInterval > 0 {
+        fetchTime = Date(timeIntervalSince1970: lastFetchTimeInterval)
+      }
+    }
+    return fetchTime
+  }
+
+  /// Last fetch status. The status can be any enumerated value from `RemoteConfigFetchStatus`.
+  @objc public var lastFetchStatus: RemoteConfigFetchStatus {
+    var currentStatus: RemoteConfigFetchStatus = .noFetchYet
+    queue.sync {
+      currentStatus = self.configFetch.settings.lastFetchStatus
+    }
+    return currentStatus
```

Comment:
Could also make use of implicit returns
```suggestion
    queue.sync {
      self.configFetch.settings.lastFetchStatus
    }
```

Reply:
Nice idea. It also enables `lastFetchTime` to become non-optional